### PR TITLE
Fine chuning functions with outputs

### PR DESCRIPTION
**Pull request by leobugeja**
Just made changes to the progress bar. Was gonna work on the prediction_summarize function but noticed the listening wasn't working properly (and saw you were debugging it)

### CODE REVIEW

File: francis/__main__.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
 import os
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"  # disables tensorflow debugging output
```
Comment (leobugeja) [inline francis/__main__.py:3]: This needs to be before all the other imports to function for somereason

File: francis/__main__.py [hunk 2]
```diff
@@ -182,7 +188,7 @@ def listen(audio_sample):
         the_model = load_model("model.h5")
 
     print("predicting ...")
-    predictions = np.around(the_model.predict(spectrograms))
+    predictions = around(the_model.predict(spectrograms))
```
Comment (leobugeja) [inline francis/__main__.py:190]: Only numpy function used in main so changed import to only import this function to reduce workload

File: francis/output_progress.py [hunk 3]
```diff
@@ -35,3 +36,7 @@ def __exit__(self, exception, value, tb):
         time.sleep(self.delay)
         if exception is not None:
             return False
+
+
+def default_bar(text, bar_length):
```
Comment (leobugeja) [inline francis/output_progress.py:41]: I know this is different from what you said about passing a bar object as an argument, however I could not think of a way to make that work as each function that uses the bar, has different parameters such as its name and its length. Therefore instead, I have created a function which returns an object, and by changing some of its defaults i.e. the fill will impact all the bars.

Comment (chrishop) [reply francis/output_progress.py:41]: What you have done is almost what I had in mind. 
I was thinking something along the lines of ...
```
processing_bar = default_bar("processing", len(pre_df.index))
pre_df = processing.process(pre_df,
                CONFIG["SAMPLE_RATE"],
                CONFIG["PREPROCESS"],
                bar_config=processing_bar)
```
This separates the logic from the view as much as possible.
But what you have done is an improvement to how it was, so its in the right direction.

One thing I'm not too big a fan of is passing in `default_bar` as the default argument, that means any situation that you **must always define a default_bar** or import when you might not want one in the first place. 

You should explicitly pass in the default bar,

 if you dont give a bar config it should default to `None`, something like this. And then have a way of handling a None type for bar_config.

Something like this ...
```
def process(df, sample_rate, pre_process=True, bar_config=None):
```

File: francis/split_filter.py [hunk 4]
```diff
@@ -4,20 +4,26 @@
 # the dataset.
 # onwards from here the data is sent to be made into a spectrogram
 
-import pandas as pd
+from pandas import DataFrame
 import numpy as np
```
Comment (chrishop) [inline francis/split_filter.py:8]: loving all the lean import will surley speed us up :)

**Review by chrishop — APPROVED** [review]

**Pull request merged**